protocol/v2/ssv/spectest/ssv_mapping_test.go

```diff
@@ -184,6 +185,19 @@ func prepareTest(t *testing.T, logger *zap.Logger, name string, test interface{}
 			Tests: typedTests,
 		}
 
+		return &runnable{
+			name: typedTest.TestName(),
+			test: func(t *testing.T) {
+				typedTest.Run(t)
+			},
+		}
+
+	case reflect.TypeOf(&runnerconstruction.RunnerConstructionSpecTest{}).String():
+		byts, err := json.Marshal(test)
+		require.NoError(t, err)
+		typedTest := &runnerconstruction.RunnerConstructionSpecTest{}
+		require.NoError(t, json.Unmarshal(byts, &typedTest))
+
```

Comment:
i think we need to duplicate the code of this test here and adapt it to call our funcs (for value checks and new runners)

you may check out some of the other spec tests for which we've already done this before

the reason we need to do this here also is because this tests the `New<Role>Runner` funcs and when we use the spec's `RunnerConstructionSpecTest` then it will only call the spec's funcs and not ours (so we're testing the spec basically)

Reply:
@moshe-blox, nice catch. I added this test to ssv and updated `ConstructBaseRunnerWithShareMap` to support the changes. Seems to be fine now